cluster-autoscaler/cloudprovider/konvoy/konvoy_manager.go

```diff
@@ -161,102 +152,84 @@ func (k *KonvoyManager) addNodeToNodeGroup(nodeGroup string) error {
-			if pool.Name == nodeGroup {
-				konvoyCluster.Spec.ProvisionerConfiguration.NodePools[i].Count++
+			if pool.Name == nodeGroupName {
+				targetPoolIndex = i
 			}
 		}
 
+		if targetPoolIndex < 0 {
+			return fmt.Errorf("node group %s does not exists", nodeGroupName)
+		}
+
+		konvoyCluster.Spec.ProvisionerConfiguration.NodePools[targetPoolIndex].Count = int32(newSize)
+
 		if err = k.dynamicClient.Update(context.Background(), konvoyCluster); err != nil {
-			klog.Warningf("Error updating the konvoy cluster: %v -- %v", konvoyCluster.Name, err)
-			err = fmt.Errorf("Failed to add node to group %s: %v", nodeGroup, err)
+			klog.Warningf("Error updating the konvoy cluster %s: %v", konvoyCluster.Name, err)
+			err = fmt.Errorf("failed to set target size %d for node group %s: %v", newSize, nodeGroupName, err)
 		} else {
-			klog.Infof("Konvoy cluster updated successfully: %v", konvoyCluster.Name)
+			klog.Infof("Konvoy %s cluster target size set to %d for node group %s", konvoyCluster.Name, nodeGroupName, newSize)
 			return nil
 		}
 	}
 
 	return err
 }
 
-// GetNodeGroupForNode returns the name of the node group to which the node
+// GetNodeGroupNameForNode returns the name of the node group to which the node
 // belongs.
-func (k *KonvoyManager) GetNodeGroupForNode(node string) (string, error) {
-	konvoyNode, err := k.getNodeByName(node)
-	if konvoyNode == nil || err != nil {
-		return "", fmt.Errorf("node %s does not exist", node)
+func (k *KonvoyManager) GetNodeGroupNameForNode(nodeName string) (string, error) {
+	kubernetesNode, err := k.getNodeByName(nodeName)
+	if kubernetesNode == nil || err != nil {
+		return "", fmt.Errorf("node %s does not exist", nodeName)
 	}
-	nodeGroup, ok := konvoyNode.Labels[nodeGroupLabel]
+	nodeGroupName, ok := kubernetesNode.Labels[nodeGroupLabel]
 	if ok {
-		return nodeGroup, nil
+		return nodeGroupName, nil
 	}
-	return "", fmt.Errorf("can't find nodegroup for node %s due to missing label %s", node, nodeGroupLabel)
+	return "", fmt.Errorf("can't find nodegroup for node %s due to missing label %s", nodeName, nodeGroupLabel)
 }
 
 func (k *KonvoyManager) getNodeByName(name string) (*apiv1.Node, error) {
-	nodes, err := k.kubeClient.CoreV1().Nodes().List(metav1.ListOptions{})
+	result, err := k.kubeClient.CoreV1().Nodes().List(metav1.ListOptions{})
 	if err != nil {
 		klog.Warningf("Error listing nodes")
 		return nil, err
 	}
 	//klog.V(2).Infof("List of nodes: %v", nodes)
-	for _, node := range nodes.Items {
-		if k.provisioner == "aws" {
-			if node.Spec.ProviderID == name {
-				klog.V(2).Infof("Get node by name: %v", name)
-				return &node, nil
-			}
-		} else {
-			if node.Name == name {
-				klog.V(2).Infof("Get node by name: %v", name)
-				return &node, nil
-			}
+	for _, node := range result.Items {
+		nodeName := kubernetesNodeName(&node, k.provisioner)
+		if nodeName == name {
+			klog.V(2).Infof("Get node by name: %v", name)
+			return &node, nil
 		}
 	}
 	return nil, nil
```

Comment:
Consider returning a NotFound error here instead of nil error to simplify [these checks](https://github.com/mesosphere/autoscaler/pull/3/files#diff-9ab55b4eceb7ec96ade6b0c2dc3c89c5R185).